forge/lang/expander.rkt

```diff
@@ -764,14 +767,45 @@
    (syntax/loc stx
     (begin
       (pred pwd.prop-name pwd.prop-expr)
-      (begin pwd.where-blocks ...)
       (test
         test_name
         #:preds [imp_total]
         #:scope pwd.scope
         #:bounds pwd.bounds
         #:expect theorem )))]))
 
+
+;; Quick and dirty static check to ensure a test
+;; references a predicate.
+
+;; Flattens test AST to a string and ensures that the
+;; referenced predicate's name is present.
+;; Could be done better by traversing the AST.
+;; This would help break apart test-expect
+;; blocks and examine each test.
+
+(define-for-syntax (ensure-target-ref target-pred)
+
+  (define tp (format "~a" (syntax->datum target-pred)))
```

Comment:
Replace with `(syntax-e target-pred)`. Then use `tp` in the `memq` and the error message.